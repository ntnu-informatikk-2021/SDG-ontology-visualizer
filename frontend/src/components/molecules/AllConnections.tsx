import { Stack } from '@chakra-ui/react';
import React from 'react';
import Connections from '../atoms/Connections';
import { Node } from '../../types/ontologyTypes';

type AllConnectionsProps = {
  contributions: Array<Node>;
  tradeOffs: Array<Node>;
  developmentAreas: Array<Node>;
  onClick: (connection: Node, predicate: Array<string>) => void;
};

// Wrapper component to render contrbutions, tradeoffs and development areas, which reduces the lines required in DetailView
const AllConnections: React.FC<AllConnectionsProps> = ({
  contributions,
  tradeOffs,
  developmentAreas,
  onClick,
}: AllConnectionsProps) => (
  <Stack spacing={5} minW="40%">
    <Connections
      connections={contributions}
      titles={['Har positiv virkning til:', 'Har ingen etablerte positive påvirkninger enda']}
      color="green"
      predicate={[
        'positiv virkning',
        'http://www.semanticweb.org/aga/ontologies/2017/9/SDG#harBidragTil',
      ]}
      handleOnClick={onClick}
    />
    <Connections
      connections={tradeOffs}
      titles={['Har negativ virkning til:', 'Har ingen etablerte negative påvirkninger enda']}
      color="red"
      predicate={[
        'negative virkning',
        'http://www.semanticweb.org/aga/ontologies/2017/9/SDG#harTradeOffTil',
      ]}
      handleOnClick={onClick}
    />
    <Connections
      connections={developmentAreas}
      titles={['Har utviklingsområde til:', 'Har ingen utviklingsområder']}
      color="blue"
      predicate={[
        'utviklingsområde',
        'http://www.semanticweb.org/aga/ontologies/2017/9/SDG#harUtviklingsOmråde',
      ]}
      handleOnClick={onClick}
    />
  </Stack>
);

export default AllConnections;
