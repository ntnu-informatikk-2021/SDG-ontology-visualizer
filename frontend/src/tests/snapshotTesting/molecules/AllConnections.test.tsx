import React from 'react';
import renderer from 'react-test-renderer';
import AllConnections from '../../../components/molecules/AllConnections';

const node1 = {
  id: 't1',
  name: 'heihei',
  type: 'ok1oki',
  prefix: {
    prefix: 'jaaajaa',
    iri: '123456',
  },
  correlation: 2,
};

const node2 = {
  id: 't2',
  name: 'hei',
  type: 'ok1',
  prefix: {
    prefix: 'jajaa',
    iri: '1234',
  },
  correlation: 3,
};

it('renders when there are no items', () => {
  const tree = renderer
    .create(
      <AllConnections contributions={[]} tradeOffs={[]} developmentAreas={[]} onClick={() => {}} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when there are one item', () => {
  const tree = renderer
    .create(
      <AllConnections
        contributions={[node2]}
        tradeOffs={[node2]}
        developmentAreas={[node2]}
        onClick={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when there are two items', () => {
  const tree = renderer
    .create(
      <AllConnections
        contributions={[node1, node2]}
        tradeOffs={[node1, node2]}
        developmentAreas={[node1, node2]}
        onClick={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
