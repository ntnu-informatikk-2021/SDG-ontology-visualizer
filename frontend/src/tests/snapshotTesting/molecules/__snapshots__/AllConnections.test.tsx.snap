// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no items 1`] = `
<div
  className="chakra-stack css-1acfu01"
>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har ingen etablerte positive påvirkninger enda
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    />
  </div>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har ingen etablerte negative påvirkninger enda
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    />
  </div>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har ingen utviklingsområder
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    />
  </div>
</div>
`;

exports[`renders when there are one item 1`] = `
<div
  className="chakra-stack css-1acfu01"
>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har positiv virkning til:
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    >
      <button
        className="chakra-button css-12ybw9u"
        onClick={[Function]}
        type="button"
      >
        hei
      </button>
    </ul>
  </div>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har negativ virkning til:
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    >
      <button
        className="chakra-button css-mmukis"
        onClick={[Function]}
        type="button"
      >
        hei
      </button>
    </ul>
  </div>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har utviklingsområde til:
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    >
      <button
        className="chakra-button css-17izuiz"
        onClick={[Function]}
        type="button"
      >
        hei
      </button>
    </ul>
  </div>
</div>
`;

exports[`renders when there are two items 1`] = `
<div
  className="chakra-stack css-1acfu01"
>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har positiv virkning til:
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    >
      <button
        className="chakra-button css-12ybw9u"
        onClick={[Function]}
        type="button"
      >
        hei
      </button>
      <button
        className="chakra-button css-aqssaa"
        onClick={[Function]}
        type="button"
      >
        heihei
      </button>
    </ul>
  </div>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har negativ virkning til:
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    >
      <button
        className="chakra-button css-mmukis"
        onClick={[Function]}
        type="button"
      >
        hei
      </button>
      <button
        className="chakra-button css-yxu8t1"
        onClick={[Function]}
        type="button"
      >
        heihei
      </button>
    </ul>
  </div>
  <b
    className="chakra-text css-1ezsviu"
  >
    Har utviklingsområde til:
  </b>
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    >
      <button
        className="chakra-button css-17izuiz"
        onClick={[Function]}
        type="button"
      >
        hei
      </button>
      <button
        className="chakra-button css-1qjqcqn"
        onClick={[Function]}
        type="button"
      >
        heihei
      </button>
    </ul>
  </div>
</div>
`;
